projects/planner/src/app/learn/learn.component.ts

```diff
@@ -85,6 +85,7 @@ export class LearnComponent implements OnInit {
         this.selectedTopic = topicName;
         this.selectedCategoryName = categoryName;
         this.activeTopic = topicName;
+        this.selectedCategory = this.findCategory(topicName, categoryName);
```

Comment:
Even better: dont store topicName and categoryName, store directly the Topic and Category.
Then it will reduce number of searches and you can also remove the this._label.